google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,341 @@
+      // neither are set, a null value will be passed to the run example method, and the example
+      // will print the hierarchies of all accessible customer IDs.
+      // params.managerId = Long.parseLong("INSERT_MANAGER_ID_HERE");
+
+      // Optional: You may pass the loginCustomerId on the command line or specify a loginCustomerId
+      // here if and only if the managerId is set. If the loginCustomerId is set neither on the
+      // command line nor below, a null value will be passed to the run example method, and the
+      // example will use each respective accessible customer ID as the loginCustomerId.
+      // params.managerId = Long.parseLong("INSERT_LOGIN_CUSTOMER_ID_HERE");
+    }
+
+    if (params.managerId != null && params.loginCustomerId == null) {
+      System.err.println("loginCustomerId must be provided if managerId is provided.");
+      return;
+    } else if (params.managerId == null && params.loginCustomerId != null) {
+      System.err.println("loginCustomerId may not be provided if managerId is not provided.");
+      return;
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetAccountHierarchy()
+          .runExample(googleAdsClient, params.managerId, params.loginCustomerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    } catch (IOException ioe) {
+      System.err.printf("Request failed. Exception: %s%n", ioe);
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param managerId the root customer ID from which to begin the search.
+   * @param loginCustomerId the loginCustomerId used to create the GoogleAdsClient.
+   * @throws IOException if a Google Ads Client is not successfully created.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, Long managerId, long loginCustomerId)
+      throws IOException {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    if (managerId != null) {
```

Comment:
nit: as a general style, I prefer leading with the positive case on if-else blocks. I just think it makes them easier to parse since it's one less boolean operator in the condition, then the not operator is naturally represented as an else block.